crates/aggkit-prover-config/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+use std::path::Path;
+use std::{
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    time::Duration,
+};
+
+use prover_config::{default_max_concurrency_limit, NetworkProverConfig, ProverType};
+use prover_logger::log::Log;
+use prover_utils::with;
+use serde::{Deserialize, Serialize};
+
+pub use crate::{shutdown::ShutdownConfig, telemetry::TelemetryConfig};
+
+pub mod shutdown;
+pub(crate) mod telemetry;
+
+pub(crate) const DEFAULT_IP: std::net::Ipv4Addr = std::net::Ipv4Addr::new(0, 0, 0, 0);
+
+/// The Aggkit Prover configuration.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub struct ProverConfig {
```

Comment:
I am confused here. Now we have duplicated `ProverConfig`, one for `agglayer-prover` and the other for `aggkit-prover`? I perfer that we start clean, and add features that we need.

Reply:
Maybe we could keep common `ProverConfig` as it is in the `prover-config` crate, and add `AgglayerProverConfig` and `AggkitProverConfig` to include things there that differ with the common field `prover_config`

Reply:
> Maybe we could keep common `ProverConfig` as it is in the `prover-config` crate, and add `AgglayerProverConfig` and `AggkitProverConfig` structures to include things there that differ with the common field `prover_config`

The `prover-*` should not link to any of `agglayer-` or `aggkit-`, it is a "framework" that can be used to compose further prover and let the community build their own if they want to. The configs of `agglayer-prover` and `aggkit-prover` will not have the same shape at all at the end.

Reply:
Ok but still lets unify `agglayer-prover-config` and `aggkit-prover-config` in the common crate. Config customizations in `aggkit-prover` and `agglayer-prover`.